src/Dscp/Crypto/Signing/Class.hs

```diff
@@ -35,18 +54,34 @@ newtype AbstractPK ss = AbstractPK (PK ss)
 deriving instance Eq (PK ss) => Eq (AbstractPK ss)
 deriving instance Ord (PK ss) => Ord (AbstractPK ss)
 deriving instance Show (PK ss) => Show (AbstractPK ss)
+deriving instance Monoid (PK ss) => Monoid (AbstractPK ss)
+
+instance (SignatureScheme ss, FromByteArray (AbstractSK ss)) =>
+         Arbitrary (AbstractSK ss) where
+    arbitrary =
+        leftToPanicWith "arbitrary abstractSK" . fromByteArray . BS.pack <$>
+        vector (secretKeyLength $ schemeLengths @ss)
 
--- | Wrapper for a secret key. 'Eq', 'Ord' and 'Show' instances
--- are not derived for security reasons.
+-- | Wrapper for a secret key. 'Show' instance is not derived for
```

Comment:
For `SignatureScheme CryptoEd25519` `Show` is safe, will show as `<scrubbed-bytes>`, http://hackage.haskell.org/package/memory-0.14.14/docs/src/Data-ByteArray-ScrubbedBytes.html#line-44

Reply:
I guess I'll remove comment then, having specific "secret" here seems more pleasant though.